playwright/tests/customers.spec.ts

```diff
@@ -0,0 +1,215 @@
+import {ADDRESS} from "@data/addresses"
+import { DeleteAddressDialog } from "@pages/dialogs/deleteAddressDialog";
+import { AddAddressDialog } from "@pages/dialogs/addAddressDialog";
+
+test.use({ storageState: "./playwright/.auth/admin.json" });
+let customersPage: CustomersPage;
+let giftCardsPage: GiftCardsPage;
+let addressesListPage: AddressesListPage;
+let addressForm: AddressForm;
+let deleteAddressDialog: DeleteAddressDialog;
+let addAddressDialog: AddAddressDialog;
+
+
+test.beforeEach(({ page }) => {
+  customersPage = new CustomersPage(page);
+  giftCardsPage = new GiftCardsPage(page);
+  addressesListPage = new AddressesListPage(page);
+  addressForm = new AddressForm(page);
+  addAddressDialog = new AddAddressDialog(page);
+  deleteAddressDialog = new DeleteAddressDialog(page);
+});
+
+test("TC: SALEOR_199 Create customer @e2e @customer", async () => {
+  const firstName = faker.name.firstName();
+  const lastName = faker.name.lastName();
+  const note = faker.lorem.sentence();
+  const email = faker.internet.email();
+  await customersPage.goToCustomersListView();
+  await customersPage.clickOnCreateCustomer();
+  await customersPage.fillFirstAndLastName(firstName, lastName);
+  await customersPage.fillEmail(email);
+  await customersPage.fillAddress(ADDRESS.addressUS);
+  await customersPage.fillNote(note);
+  await customersPage.saveCustomer();
+  await customersPage.successBanner.waitFor({ state: "visible" });
+  await expect(customersPage.pageHeader).toContainText(
+    `${firstName} ${lastName}`,
+  );
+  await expect(customersPage.customerNoteInput).toContainText(note);
+  await expect(customersPage.customerFirstNameInput).toHaveValue(firstName);
+  await expect(customersPage.customerLastNameInput).toHaveValue(lastName);
+  await expect(customersPage.customerEmailInput).toHaveValue(
+    email.toLowerCase(),
+  );
+});
+test("TC: SALEOR_200 As an admin I should not be able to create customer with duplicated email @e2e @customer", async () => {
+  const firstName = faker.name.firstName();
+  const lastName = faker.name.lastName();
+  const note = faker.lorem.sentence();
+  await customersPage.goToCustomersListView();
+  await customersPage.clickOnCreateCustomer();
+  await customersPage.fillFirstAndLastName(firstName, lastName);
+  await customersPage.fillEmail(CUSTOMERS.customerToBeDeactivated.email);
+  await customersPage.fillAddress(ADDRESS.addressUS);
+  await customersPage.fillNote(note);
+  await customersPage.saveCustomer();
+  await customersPage.expectErrorBannerMessage("User with this Email already exists.")
+});
+
+test("TC: SALEOR_201 Update customer account info @e2e @customer", async () => {
+  const firstName = faker.name.firstName();
+  const lastName = faker.name.lastName();
+  const email = faker.internet.email();
+  const note = faker.lorem.sentence();
+  await customersPage.waitForNetworkIdle(() =>
+    customersPage.gotoCustomerDetailsPage(CUSTOMERS.editCustomer.id),
```

Comment:
Can we wait for element instead? Mixing layers is sth I am always worry about.

Reply:
@wojteknowacki I did not make the change you requested as I do not agree. 
I've added waitForNetworkIdle() function in order to tackle slow network issues on our CI - the page was still loading and elements were not visible on the page due to which the tests failed. I think waiting for the request to finish is the safest and the most e2e approach we can rely on.

Reply:
I get it :) how about including waitForNetworkIdle in the gotoCustomerDetailsPage definition? This way if sth changes you have to update only the definition.

Reply:
Plus the test definition is cleaner then ;)

Reply:
done